ingest/build-configs/ncbi/rules/curate.smk

```diff
@@ -0,0 +1,167 @@
+"""
+This part of the workflow handles the curation of data from NCBI
+
+REQUIRED INPUTS:
+
+    ndjson      = ncbi/data/ncbi.ndjson
+
+OUTPUTS:
+
+    metadata    = ncbi/results/metadata.tsv
+    seuqences   = ncbi/results/sequences_{segment}.fasta
+
+"""
+
+
+rule fetch_general_geolocation_rules:
+    output:
+        general_geolocation_rules="ncbi/data/general-geolocation-rules.tsv",
+    params:
+        geolocation_rules_url=config["curate"]["geolocation_rules_url"],
+    shell:
+        """
+        curl {params.geolocation_rules_url} > {output.general_geolocation_rules}
+        """
+
+
+rule concat_geolocation_rules:
+    input:
+        general_geolocation_rules="ncbi/data/general-geolocation-rules.tsv",
+        local_geolocation_rules=config["curate"]["local_geolocation_rules"],
+    output:
+        all_geolocation_rules="ncbi/data/all-geolocation-rules.tsv",
+    shell:
+        """
+        cat {input.general_geolocation_rules} {input.local_geolocation_rules} >> {output.all_geolocation_rules}
+        """
+
+
+def format_field_map(field_map: dict[str, str]) -> str:
+    """
+    Format dict to `"key1"="value1" "key2"="value2"...` for use in shell commands.
+    """
+    return " ".join([f'"{key}"="{value}"' for key, value in field_map.items()])
+
+
+rule curate:
+    input:
+        sequences_ndjson="ncbi/data/ncbi.ndjson",
+        all_geolocation_rules="ncbi/data/all-geolocation-rules.tsv",
+        annotations=config["curate"]["annotations"],
+    output:
+        curated_ndjson="ncbi/data/curated.ndjson",
+    log:
+        "ncbi/logs/curate.txt",
+    benchmark:
+        "ncbi/benchmarks/curate.txt"
+    params:
+        field_map=format_field_map(config["curate"]["field_map"]),
+        strain_regex=config["curate"]["strain_regex"],
+        strain_backup_fields=config["curate"]["strain_backup_fields"],
+        date_fields=config["curate"]["date_fields"],
+        expected_date_formats=config["curate"]["expected_date_formats"],
+        articles=config["curate"]["titlecase"]["articles"],
+        abbreviations=config["curate"]["titlecase"]["abbreviations"],
+        titlecase_fields=config["curate"]["titlecase"]["fields"],
+        authors_field=config["curate"]["authors_field"],
+        authors_default_value=config["curate"]["authors_default_value"],
+        abbr_authors_field=config["curate"]["abbr_authors_field"],
+        segments=format_field_map(config["ncbi_segments"]),
+        host_map=config["curate"]["host_map"],
+        annotations_id=config["curate"]["annotations_id"],
+    shell:
+        """
+        (cat {input.sequences_ndjson} \
+            | ./vendored/transform-field-names \
+                --field-map {params.field_map} \
```

Comment:
Noted by @jameshadfield on Slack

This outputs a _lot_ of warning messages when for column names that are the same. Can use the `--force` flag to silence the warnings.

Reply:
Resolving upstream in https://github.com/nextstrain/ingest/pull/38. 

Will update the vendored scripts here once it's merged.

Reply:
Merged and updated vendored scripts in https://github.com/nextstrain/avian-flu/pull/40/commits/a2b01e7e2b351fa067b8ad694481e10cd872318d.